### PR TITLE
fix(tests) : Update UI tests to run on both Agile and SDD template parallelly

### PR DESCRIPTION
**Pull request by Raunak1203**
<strong>Note: If there are pending changes to the PR, prefix the PR title with "WIP" and add the label "DO NOT MERGE"</strong>

### Mandatory
 - [x] What does this PR do? 
This PR -
- Update UI tests to run on both Agile and SDD template parallelly.
- Tests which are template-dependent will be part of sddTemplate.spec and agileTemplate.spec.ts and will not be run parallelly.
- Constant.ts contains data specific to the template.

- [ ] What issue/task does this PR references?

- [ ] Are the tests Included?
___
### Optional
- [ ] Is the documentation Included?

- [ ] Are the Release Notes included?

- [ ] @mention(s) to expected reviewer(s) included

### CODE REVIEW

**Comment by Raunak1203** [comment]
[test]

**Comment by Raunak1203** [comment]
[test]

**Comment by Raunak1203** [comment]
[test]

**Comment by Raunak1203** [comment]
[test]

**Comment by Raunak1203** [comment]
[test]

**Comment by Raunak1203** [comment]
[test]

**Comment by Raunak1203** [comment]
[test]

**Comment by Raunak1203** [comment]
[test]

**Comment by Raunak1203** [comment]
[test]

**Comment by Raunak1203** [comment]
[test]

**Comment by Raunak1203** [comment]
[test]

**Comment by Raunak1203** [comment]
[test]

**Comment by Raunak1203** [comment]
[test]

**Comment by Raunak1203** [comment]
[test]

**Comment by Raunak1203** [comment]
[test]

**Comment by Raunak1203** [comment]
[test]

File: src/tests/ui/base.element.ts [hunk 1]
```diff
@@ -58,25 +59,45 @@ export class BaseElement extends ElementFinder implements BaseElementInterface {
   }
 
   async untilClickable(timeout?: number) {
-    await this.waitFor('clickable', EC.elementToBeClickable(this), timeout);
+    try {
+      await this.waitFor('clickable', EC.elementToBeClickable(this), timeout);
+    } catch (e) {
+      this.fail('ERROR Element: ', this.name, 'is not clickable');
```
Comment (jarifibrahim) [inline src/tests/ui/base.element.ts]: Missing whitespace after `this.name`

File: src/tests/ui/base.element.ts [hunk 2]
```diff
@@ -58,25 +59,45 @@ export class BaseElement extends ElementFinder implements BaseElementInterface {
...
+    }
   }
 
   async untilPresent(timeout?: number) {
-    await this.waitFor('present', EC.presenceOf(this), timeout);
+    try {
+      await this.waitFor('present', EC.presenceOf(this), timeout);
+    } catch (e) {
+      this.fail('ERROR Element: ', this.name, 'is not present');
```
Comment (jarifibrahim) [inline src/tests/ui/base.element.ts]: Missing whitespace after `this.name`

File: src/tests/ui/base.element.ts (hunk 1 above)
Comment (jarifibrahim) [reply src/tests/ui/base.element.ts]: ~Missing whitespace after this.name~

File: src/tests/ui/base.element.ts (hunk 2 above)
Comment (jarifibrahim) [reply src/tests/ui/base.element.ts]: ~Missing whitespace after this.name~

File: src/tests/specs/detailPage.spec.ts [hunk 3]
```diff
@@ -6,13 +6,14 @@ import * as support from '../support';
...
   let planner: PlannerPage;
   let c = new support.Constants();
+  let testData;
 
   beforeAll(async () => {
     await support.desktopTestSetup();
     planner = new PlannerPage(browser.baseUrl);
     await planner.openInBrowser();
     await planner.waitUntilUrlContains('typegroup');
-    await planner.ready();
```
[the remark this answers is not in the document]
Comment (Raunak1203) [reply src/tests/specs/detailPage.spec.ts]: @jarifibrahim detailPage invloves opening opening detail page. So, I have included await planner.ready() in beforeEach()

File: src/tests/mixins/logging.ts [hunk 4]
```diff
@@ -1,15 +1,22 @@
...
   debug(context: string, ...msg: string[]) {
     let className = this.constructor.name;
-    debug(`... ${className}('${this.name}'): ${context}`, ...msg);
+    debug(`${browser.browserName} ${className}('${this.name}'): ${context}`, ...msg);
+  }
+
+  fail(action: string, ...msg: string[]) {
+    let className = this.constructor.name;
+    /* to display errors in red */
+    console.error('\x1b[31m%s', `${action}: ${browser.browserName} ${className}('${this.name}')`, ...msg);
```
[the remark this answers is not in the document]
Comment (Raunak1203) [reply src/tests/mixins/logging.ts]: done https://github.com/fabric8-ui/fabric8-planner/pull/2747/commits/5c17f4ae27d96877aacda0795075167b280b002f#diff-d0ebeccdc97f4907f608dc5475aa5229R20

File: src/tests/specs/detailPage.spec.ts [hunk 5]
```diff
@@ -52,71 +54,74 @@ describe('Detail View test: ', () => {
...
-      Workitem_Title_3 = 'Workitem_Title_3';
-    await planner.workItemList.openDetailPage(c.workItemTitle2);
+      workItemTitle2 = 'Workitem_Title_2';
+    await planner.workItemList.openDetailPage(workItemTitle2);
     await planner.waitUntilUrlContains('detail');
-    await planner.detailPage.titleInput.untilTextIsPresentInValue(c.workItemTitle2);
-    await planner.detailPage.addLink(linkType, searchWorkItem, Workitem_Title_3);
-    expect(await planner.detailPage.getLinkedItems()).toContain(Workitem_Title_3);
+    await planner.detailPage.titleInput.untilTextIsPresentInValue(workItemTitle2);
+    await planner.detailPage.addLink(linkType, testData.searchWorkItem3, testData.Workitem_Title_3);
```
[the remark this answers is not in the document]
Comment (Raunak1203) [reply src/tests/specs/detailPage.spec.ts]: done https://github.com/fabric8-ui/fabric8-planner/pull/2747/commits/5c17f4ae27d96877aacda0795075167b280b002f#diff-d0ebeccdc97f4907f608dc5475aa5229R20

File: src/tests/specs/detailPage.spec.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (jarifibrahim) [reply src/tests/specs/detailPage.spec.ts]: okay

**Pull request merged**